RadixWallet/Features/AssetsFeature/AssetsView+Update.swift

```diff
@@ -0,0 +1,178 @@
+				let resourceDetails = state.accountPortfolio.poolUnitDetails.flatten().flatMap {
+					$0.first { poolUnit.resourcePoolAddress == $0.address }.map(Loadable.success) ?? .loading
+				}
+				let updatedPoolUnit = state.resources.poolUnitsList?.update(poolUnit: poolUnit, resourceDetails: resourceDetails)
+
+				return updatedPoolUnit ?? PoolUnitsList.State.PoolUnitState(
+					poolUnit: poolUnit,
+					resourceDetails: resourceDetails,
+					isSelected: mode.nonXrdRowSelected(poolUnit.resource.resourceAddress)
+				)
+			}.asIdentified()
+		)
+
+		let stakes = portfolio.account.poolUnitResources.radixNetworkStakes
+
+		let stakeUnitList: StakeUnitList.State? = {
+			guard !stakes.isEmpty else { return nil }
+
+			let stakeUnitDetails = state.accountPortfolio.stakeUnitDetails.flatten()
+			if let stakeUnitList = state.resources.stakeUnitList {
+				return .init(
+					account: stakeUnitList.account,
+					selectedLiquidStakeUnits: stakeUnitList.selectedLiquidStakeUnits,
+					selectedStakeClaimTokens: stakeUnitList.selectedStakeClaimTokens,
+					stakeUnitDetails: stakeUnitDetails
+				)
+			} else {
+				return .init(
+					account: portfolio.account,
+					selectedLiquidStakeUnits: mode.selectedAssets.map { assets in
+						let stakeUnitResources = stakes.map(\.stakeUnitResource)
+						return assets
+							.fungibleResources
+							.nonXrdResources
+							.filter(stakeUnitResources.contains)
+							.asIdentified()
+					},
+					selectedStakeClaimTokens:
+					mode.isSelection ?
+						stakes
+						.compactMap(\.stakeClaimResource)
+						.reduce(into: StakeUnitList.SelectedStakeClaimTokens()) { dict, resource in
+							if let selectedtokens = mode.nftRowSelectedAssets(resource.resourceAddress)?.elements.asIdentified() {
+								dict[resource] = selectedtokens
+							}
+						} : nil,
+					stakeUnitDetails: stakeUnitDetails
+				)
+			}
+		}()
+
+		state.totalFiatWorth.refresh(from: portfolio.totalFiatWorth)
+		state.resources = .init(
+			fungibleTokenList: fungibleTokenList,
+			nonFungibleTokenList: !nfts.isEmpty ? .init(rows: nfts.asIdentified()) : nil,
+			stakeUnitList: stakeUnitList,
+			poolUnitsList: poolUnitList
+		)
+	}
+}
+
+extension AccountPortfoliosClient.AccountPortfolio {
+	mutating func refresh(from portfolio: AccountPortfoliosClient.AccountPortfolio) {
+		self.account = portfolio.account
+		self.isCurrencyAmountVisible = portfolio.isCurrencyAmountVisible
+		self.fiatCurrency = portfolio.fiatCurrency
+		self.stakeUnitDetails.refresh(from: portfolio.stakeUnitDetails)
+		self.poolUnitDetails.refresh(from: portfolio.poolUnitDetails)
+	}
+}
+
+extension Loadable where Value == AccountPortfoliosClient.AccountPortfolio {
+	mutating func refresh(from portfolio: Loadable<Value>) {
+		self.refresh(from: portfolio, valueChangeMap: { old, new in
+			var old = old
+			old.refresh(from: new)
+			return old
+		})
+	}
+}
+
+extension FungibleAssetList.State {
+	public mutating func update(
```

Comment:
These `update` functions should rather be called `updated`.

Additionally, the functions which update a row, need to be named `updatedRow`